server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala

```diff
@@ -302,6 +307,20 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
     case _ => Left(new IllegalArgumentException("Expecting htmlBody to contains only 1 part"))
   }
 
+  def validateTextBody: Either[IllegalArgumentException, Option[String]] = textBody match {
+    case None => Right(None)
+    case Some(html :: Nil) if !html.`type`.value.equals("text/plain") => Left(new IllegalArgumentException("Expecting htmlBody type to be text/html"))
+    case Some(html :: Nil) => bodyValues.getOrElse(Map())
+      .get(html.partId)
+      .map {
+        case part if part.isTruncated.isDefined && part.isTruncated.get.value.equals(true) => Left(new IllegalArgumentException("Expecting isTruncated to be false"))
+        case part if part.isEncodingProblem.isDefined && part.isEncodingProblem.get.value.equals(true) => Left(new IllegalArgumentException("Expecting isEncodingProblem to be false"))
+        case part => Right(Some(part.value))
+      }
+      .getOrElse(Left(new IllegalArgumentException("Expecting bodyValues to contain the part specified in textBody")))
```

Comment:
A bit verbose for this case i think. Could you extract this?

Reply:
You've been the one writting that verbose piece of code on the first place ;-)

(But I'm guilty of a copy & paste)